tests.js

```diff
@@ -1787,7 +1794,7 @@ QUnit.test("Error Messages", function(t) {
 
 QUnit.test("Regression Tests", function(t) {
   this.assert_equals('make "x 0  repeat 3 [ for [ i 1 4 ] [ make "x :x + 1 ] ]  :x', 12);
-  this.assert_equals('make "x 0  for [i 0 100 :i + 1] [make "x :x + :i]  :x', 120);
```

Comment:
Interesting... this starts failing ("don't know about variable i").

In UCBLogo it yields 5050 (which seems correct; not sure where 120 came from)

Also, the regression tests hang. I'll take a closer look.

Reply:
Sorry, hang is only if I introduce `setvar(varname, current);` into the initial block after `current` is calculated. That causes the new test to hang, since then `i` becomes 0. So that's not the right tweak...

Reply:
... and my testing in UCBLogo was incorrect, since it parses differently - it's not reparsing the tail of the list as an expression but simply evaluating the 4th member. It also declares a local scope which is _not_ used when evaluating the control block, e.g.

```
make "i 2
for [ i 0 10 :i ] [ show :i]
```
yields 0,2,4,5,6,8,10